src/librustc_codegen_ssa/traits/intrinsic.rs

```diff
@@ -27,11 +31,16 @@ pub trait IntrinsicCallMethods<'tcx>: HasCodegen<'tcx> {
         llresult: Self::Value,
         span: Span,
     );
-}
 
-pub trait IntrinsicDeclarationMethods<'tcx>: Backend<'tcx> {
-    fn get_intrinsic(&self, key: &str) -> Self::Value;
+    fn abort(&mut self);
+    fn assume(&mut self, val: Self::Value);
+    fn expect(&mut self, cond: Self::Value, expected: bool) -> Self::Value;
 
-    /// Declare any llvm intrinsics that you might need
-    fn declare_intrinsic(&self, key: &str) -> Option<Self::Value>;
+    fn call_overflow_intrinsic(
```

Comment:
I'd move this to the builder and treat it like a regular instruction (I assume that's how Cranelift would handle it).